### DISCUSSION TITLE
3rd device for nav directions

### DISCUSSION THREAD

**Discussion by Zerberuss** [BimmerGestalt/AAIdrive#583] [Ideas]
Hello!
First the app is amazing and I love it!!!

Since the old connected App was removed from the store and Spotify never worked reliable with my iPhone, I bought an old Android only for AAidrive to get Spotify in my car. With some tasker profiles and rooting to shut down and reboot my Android this works perfectly.

Are there any ideas how to send nav directions from my iPhone to AAdrive, which transfers them over to the build in nav system automatically?

(I like the build in nav most, because it uses the heads-up display)

**Comment by hufman** [comment]
Thank you for the kind compliments!
That is a clever idea, but I don't know any existing solutions to do that. The actual navigation integration with the car uses the standard [Android Navigation Intents](https://stackoverflow.com/a/59598183/169035) system (and here's several [map url parsing examples](https://github.com/BimmerGestalt/AAIdrive/blob/598d5a79b9363a1bc4f8416c435ba80ba6ec648e/app/src/androidTest/java/me/hufman/androidautoidrive/NavigationParserTest.kt)), so it would work as long as you can generate and send such an Intent on the phone. It looks like Tasker can natively send Intents, so you can test it out that way. I'm not sure what would provide the next level, to have an app on the iPhone send a destination to Android to be converted into a Navigation Intent.

**Comment by Zerberuss** [comment]
Hello!

I found a solution that works for me.
I add a calendar entry with the location, which gets synced to the phone in the car. With the MyCalendar app i can open the entry and start the navigation to the destination.
I figured out that somehow sometimes the navigation to the next appointment Starts automatically, but not always.
I tested it with entries which were always 5min from now when i started the car, but only sometimes the car figured it out automatically.
Not a bummer but is there a fix for that? Do i miss something?

**Reply by hufman** [reply]
That's a good workaround!
I also sometimes see that bug in the auto navigation, and figured it's just because of poor internet in my neighborhood as I'm starting out on a drive. Do you also have poor internet, or is there some deeper bug?

**Reply by Zerberuss** [reply]
I don't know when the automatically navigation should be triggered. 
Poor internet shouldn't be the problem where I live. 
Because of my setup, with the phone booting which the ignition of the car, its possible that my calendar gets synced after the AAIdrive app has started.

**Reply by hufman** [reply]
It should try to navigate a few seconds after the Calendar car app is [created](https://github.com/BimmerGestalt/AAIdrive/blob/main/app/src/main/java/me/hufman/androidautoidrive/carapp/calendar/CalendarAppService.kt#L44), but indeed I'm not sure when the phone's calendar gets synced with the cloud.
